processing/src/main/java/io/druid/query/groupby/GroupByQueryQueryToolChest.java

```diff
@@ -426,4 +505,22 @@ public Row apply(Object input)
       }
     };
   }
+
+
+  public static Set<String> extractionsToRewrite(GroupByQuery query, final boolean getPre)
+  {
+    final Set<String> rewriteSet = new HashSet<>();
+    for (final DimensionSpec spec : query.getDimensions()) {
+      if (
+          spec != null
+          && spec.getExtractionFn() != null
+          && ExtractionFn.ExtractionType.ONE_TO_ONE.equals(
```

Comment:
can we comment why only ONE_TO_ONE are included here? It took me a little bit to figure out that these changes are an optimization and only applies to renaming.